### DISCUSSION TITLE
Access to swup instance from astro integration

### DISCUSSION THREAD

**Discussion by krishnaTORQUE** [swup/swup#914] [Help] [answered]
How to redirect or navigate user in svelte?

**Comment by daun** [comment]
Care to clarify the question? This is a bit cryptic 🕳️

**Comment by krishnaTORQUE** [comment]
I am using svelte in astro.
I added swup by `npx astro add`.
Everything is working with animation but 1 problem is.
How can I navigate/redirect from 1 page to another?

E.g.
`navigate("/home")`

But there is no navigate function I can find.

**Reply by daun** [reply, marked as answer]
You can access `swup.navigate` from a [global instance](https://github.com/swup/astro?tab=readme-ov-file#access-to-the-swup-instance).

**Reply by krishnaTORQUE** [reply]
> You can access `swup.navigate` from a [global instance](https://github.com/swup/astro?tab=readme-ov-file#access-to-the-swup-instance).

Thanks a lot. It worked

**Comment by krishnaTORQUE** [comment]
So basically this api is not available in svelte or in astro ?


<img width="912" alt="Screenshot 2024-06-28 at 2 42 28 AM" src="https://github.com/swup/swup/assets/4960568/33278d21-da0e-4851-b9c3-eac5b17e81f4">

**Reply by hirasso** [reply]
Yes, it is. You just have to use `window.swup` and have `globalInstance` set to true in the options.
